packages/opentelemetry-sdk-setup/src/util.ts

```diff
@@ -0,0 +1,122 @@
+import { createWriteStream } from 'node:fs'
+import { realpath, readFile } from 'node:fs/promises'
+
+import { diag, context, DiagLogger, Context } from '@opentelemetry/api'
+import { parseKeyPairsIntoRecord } from '@opentelemetry/core/build/src/baggage/utils.js'
+import { readPackageUp, PackageJson } from 'read-pkg-up'
+
+/**
+ * Builds a function for logging data to a provided fileDescriptor (i.e. hidden from
+ * the user-facing build logs)
+ * This has been pulled from @netlify/build as a quick way to hook into the system logger
+ */
+const getSystemLogger = function (
+  debug: boolean,
+  /** A system log file descriptor, if non is provided it will be a noop logger */
+  systemLogFile?: number,
+): (...args: any[]) => void {
+  // If the `debug` flag is used, we return a function that pipes system logs
+  // to the regular logger, as the intention is for them to end up in stdout.
+  // For now we just use plain `console.log`, later on we can revise it
+  if (debug) {
+    return (...args) => console.log(...args)
+  }
+
+  // If there's not a file descriptor configured for system logs and `debug`
+  // is not set, we return a no-op function that will swallow the errors.
+  if (!systemLogFile) {
+    return () => {
+      // no-op
+    }
+  }
+
+  // Return a function that writes to the file descriptor configured for system
+  // logs.
+  const fileDescriptor = createWriteStream('', { fd: systemLogFile })
+
+  fileDescriptor.on('error', () => {
+    console.error('Could not write to system log file')
+  })
+
+  return (...args) => fileDescriptor.write(`${console.log(...args)}\n`)
+}
+
+/** Given a simple logging function return a `DiagLogger`. Used to setup our system logger as the diag logger.*/
+export const getOtelLogger = function (
```

Comment:
nit: The type is `DiagLogger` but the name of the function is `getOtelLogger`. Would it make sense to unify that?

Reply:
Done via - https://github.com/netlify/build/pull/5388/commits/1e5e44eedd8510b78b1be5c9ef6367569b445b3c